src/Client/DTO/WorkflowExecutionInfo.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Temporal\Client\DTO;
+
+use DateTimeInterface;
+use JetBrains\PhpStorm\Immutable;
+use Temporal\Client\DTO\ResetPointInfo as ResetPointInfoDto;
+use Temporal\DataConverter\EncodedCollection;
+use Temporal\DataConverter\ValuesInterface;
+use Temporal\Workflow\WorkflowExecution;
+use Temporal\Workflow\WorkflowType;
+
+/**
+ * @psalm-immutable
+ */
+#[Immutable]
+final class WorkflowExecutionInfo
+{
+    /**
+     * @param array<ResetPointInfoDto> $autoResetPoints
+     */
+    public function __construct(
+        public WorkflowExecution $execution,
+        public WorkflowType $type,
+        public ?DateTimeInterface $startTime,
+        public ?DateTimeInterface $closeTime,
+        public int $status,
+        public int $historyLength,
+        public ?string $parentNamespaceId,
+        public ?WorkflowExecution $parentExecution,
+        public ?DateTimeInterface $executionTime,
+        public EncodedCollection $memo,
+        public EncodedCollection $searchAttributes,
+        public array $autoResetPoints,
+        public string $taskQueue,
+        public int $stateTransitionCount,
+    ) {
```

Comment:
This is also missing the recently added version stamp: https://github.com/temporalio/api/blob/5f21c72de0ec9e6037e84c95f097641b71d7c39f/temporal/api/workflow/v1/message.proto#L61